javascript/ql/lib/semmle/javascript/frameworks/CryptoLibraries.qll

```diff
@@ -316,9 +343,31 @@ private module CryptoJS {
       this = getDirectApplication(input, algorithm)
     }
 
-    override DataFlow::Node getInput() { result = input }
+    override DataFlow::Node getAnInput() { result = input }
 
     override CryptographicAlgorithm getAlgorithm() { result = algorithm }
+
+    // e.g. CryptoJS.AES.encrypt("msg", "key", { mode: CryptoJS.mode.<modeString> })
+    private BlockMode getExplicitBlockMode() {
+      exists(DataFlow::ObjectLiteralNode o, DataFlow::SourceNode modeNode, string modeString |
```

Comment:
```suggestion
      exists(DataFlow::SourceNode o, DataFlow::SourceNode modeNode, string modeString |
```

Reply:
Actually, instead of this I think it would be better to use more API-graphs instead.   
I tried it out, it worked, and I took the liberty of pushing it straight to this PR.